docs/HACKING.md

```diff
@@ -187,29 +187,51 @@ REPO={docker.io/username} ./hack/push-image.sh
 Quay.io or any other public registry isn't strictly required - you can use a local
 registry as long as those images are pullable.
 
+By default, podman created quay.io repo is marked as private, make sure to
+change it in the quay.io webapge to public, allowing the cluster to pull
+the update payload.
+
 ## Build a custom release payload
 
+Get a pull secret from https://console.redhat.com/openshift/install/pull-secret
+and store it to `$HOME/.kube/pull-secret.txt` or any place you like.
+
+Besides pull secret, you need to run `podman login quay.io:443` and `oc
+registry login`. The reason to use `quay.io:443` is because pull secret already
+has `quay.io` login information for
+`quay.io/openshift-release-dev/ocp-release`, but you need your personal account
+to upload to `quay.io:443/user/origin-release`.
+
 Now that your have your custom image, to build a custom release payload, run:
 
 ```
-oc adm release new -n origin --server https://api.ci.openshift.org \
-                                --from-image-stream "{version number}" \
-                                --to-image quay.io/user/origin-release:v{version number} \
-                                machine-config-operator=quay.io/user/machine-config-operator:latest
+oc adm release new \
+    -a ~/.kube/pull-secret.txt \
+    --from-release \
+    quay.io/openshift-release-dev/ocp-release:{version_number}-x86_64 \
+    --name {version_number} \
+    --to-image quay.io:443/<your_user>/origin-release:v{version_number} \
```

Comment:
Does specifying port is must?

Reply:
it is, because you have different credentials for pulling quay.io/openshift-release-dev/ocp-release and for pushing to your own quay repo. So you register one with quay.io:443 so podman knows which credential to use for which image. 
Otherwise both are quay.io and you can't have two credentials for the same address.